Lib/fontTools/varLib/instancer/__init__.py

```diff
@@ -1273,9 +1271,57 @@ def instantiateVariableFont(
         },
     )
 
+    # Set Regular/Italic/Bold/Bold Italic bits as appropriate, after the name
+    # table has been updated.
+    setRibbiBits(varfont)
```

Comment:
I wonder if you should do it only when updateFontNames is True, that's the thing that may or may not trigger changes in style name. If the original varfont that is being instanced didn't have those bit sets correctly it's not our job to fix them here. Only if the style name changed there's a chance these bits will need to be updated.